lib/datasource/git-refs/index.ts

```diff
@@ -32,23 +32,35 @@ export async function getRawRefs({
     }
 
     // fetch remote tags
-    const lsRemote = await git.listRemote([lookupName, '--sort=-v:refname']);
+    const lsRemote = await git.listRemote([
+      '--sort=-v:refname',
+      '--tags',
+      '--heads',
+      '--refs',
+      lookupName,
+    ]);
 
     if (!lsRemote) {
       return null;
     }
 
-    const refs = lsRemote.replace(/^.+?refs\//gm, '').split('\n');
+    const refs = lsRemote
+      .trim()
+      .replace(/^.+?refs\//gm, '')
+      .split('\n');
 
-    const result = refs.map(ref => ({
-      type: /(.*?)\//.exec(ref)[1],
-      value: /\/(.*)/.exec(ref)[1],
-    }));
+    const result = refs.map(ref => {
+      const match = /(?<type>\w+)\/(?<value>.*)/.exec(ref);
```

Comment:
perf: move regex to const in module scope

Reply:
Rather odd request considering this has not been requested on the initial implementation, but i have extracted all regexes to const for git-refs and git-tags ;)

Reply:
sorry, havn't seen it before. 🙈 😅